spring-data-jdbc/src/test/java/org/springframework/data/jdbc/core/convert/DefaultDataAccessStrategyUnitTests.java

```diff
@@ -76,11 +78,13 @@
 	JdbcConverter converter;
 	DefaultDataAccessStrategy accessStrategy;
 
+	private Dialect dialect;
+
 	@BeforeEach
 	public void before() {
 
 		DelegatingDataAccessStrategy relationResolver = new DelegatingDataAccessStrategy();
-		Dialect dialect = HsqlDbDialect.INSTANCE;
+		dialect = HsqlDbDialect.INSTANCE;
```

Comment:
If we move the `dialect` into a field we should also the initialisation with it.
But it would be easier to hard code the exact expected insert statement and leave the `Dialect` where it is.

Reply:
Resolved with [this](https://github.com/spring-projects/spring-data-jdbc/pull/1047/commits/71204add3936c7a8ee171d01b68ffe4f8a3d7b32)